### PR TITLE
REF: use _get_string_slice in PeriodIndex.get_value

### PR DESCRIPTION
**Pull request by jbrockmendel**
Most of _get_string_slice is duplicated inside PeriodIndex.get_value, which likely explains why get_string_slice has almost no test [coverage](https://codecov.io/gh/pandas-dev/pandas/src/master/pandas/core/indexes/period.py#L715).  This also brings PeriodIndex.get_value into closer alignment with DatetimeIndex.get_value, so we may be able to share code a few steps from here.

### CODE REVIEW

File: pandas/core/indexes/period.py [hunk 1]
```diff
@@ -671,60 +666,38 @@ def _maybe_cast_slice_bound(self, label, side, kind):
...
     def _get_string_slice(self, key):
+        raw = key
         if not self.is_monotonic:
             raise ValueError("Partial indexing only valid for ordered time series")
 
         key, parsed, reso = parse_time_string(key, self.freq)
         grp = resolution.Resolution.get_freq_group(reso)
         freqn = resolution.get_freq_group(self.freq)
-        if reso in ["day", "hour", "minute", "second"] and not grp < freqn:
-            raise KeyError(key)
```
Comment (jbrockmendel) [inline pandas/core/indexes/period.py]: @jreback i think you wrote most of this, do you remember what the reso checks here are for?

Comment (jreback) [reply pandas/core/indexes/period.py]: nope

**Comment by jreback** [comment]
needs a rebase, this overalaps with #31096 ?

**Comment by jbrockmendel** [comment]
> needs a rebase, this overalaps with #31096 ?

rebased+green.  the diff overlaps so 31096 will probably need to be rebased, but they are logically independent.

**Review by jreback — APPROVED** [review]
lgtm merge on green.

**Pull request merged**